sources/corepackUtils.ts

```diff
@@ -137,16 +148,26 @@ export async function installVersion(installTarget: string, locator: Locator, {s
 
   stream.pipe(sendTo);
 
-  const hash = build[0]
-    ? stream.pipe(createHash(build[0]))
-    : null;
-
+  const hash = stream.pipe(createHash(build[0] ?? `sha256`));
   await once(sendTo, `finish`);
 
-  const actualHash = hash?.digest(`hex`);
-  if (actualHash !== build[1])
+  const actualHash = hash.digest(`hex`);
+  if (build[1] && actualHash !== build[1])
     throw new Error(`Mismatch hashes. Expected ${build[1]}, got ${actualHash}`);
 
+  await fs.promises.writeFile(path.join(tmpFolder, `.corepack`), JSON.stringify({
+    locator,
+    hash: actualHash,
```

Comment:
```suggestion
    hash: `sha224.${actualHash}`,
```

Reply:
Don't forget to merge this, or I guess the SHA-256 variant if we're not using SHA-224.
```suggestion
    hash: `sha256.${actualHash}`,
```